packages/DataTable/src/hooks/useBestTableDimensions.ts

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import debounce from "lodash.debounce";
+import convertSizeStringToNumber, { Direction } from "../helpers/convertSizeStringToNumber";
+
+type Dimensions = {
+  width: number;
+  height: number;
+  shouldHaveHorizontalScroll: boolean;
+};
+
+export default function useBestTableDimensions({
+  tableRef,
+  componentRef,
+  height: maxHeight,
+  width: maxWidth,
+  shouldResizeWithViewport,
+}: {
+  tableRef: React.RefObject<HTMLDivElement>;
+  componentRef: React.RefObject<unknown>;
+  height: string;
+  width: string;
+  shouldResizeWithViewport: boolean;
+}): Dimensions {
+  const [dimensions, setDimensions] = React.useState<Dimensions>({
```

Comment:
Maybe we could use `() => ` syntax here, so `convertSizeStringToNumber` is not called on every render redundantly